auslib/web/base.py

```diff
@@ -105,3 +105,8 @@ def robots():
     view_func=ClientRequestView.as_view('clientrequest_esrnightly'),
     defaults={'queryVersion': 3},
 )
+app.add_url_rule(
+    '/update/4///////////update.xml',
+    view_func=ClientRequestView.as_view('clientrequest4_unsubstituted'),
+    defaults={'queryVersion': 4},
+)
```

Comment:
This isn't like the URLs described in the bug. The ones we need to handle are:
/update/1/%PRODUCT%/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/update.xml
/update/2/%PRODUCT%/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/update.xml
/update/3/%PRODUCT%/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/update.xml
/update/4/%PRODUCT%/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/%MOZ_VERSION%/update.xml
/update/6/%PRODUCT%/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%SYSTEM_CAPABILITIES%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/update.xml

And they should get routed differently - there's no point in putting them through ClientRequestView. You can probably just add the routes with @app.route (like we do for robots.txt, and then return a 404 from there. This avoids running _any_ database code at all, which should speed these queries up.